plan/new_physical_plan_builder.go

```diff
@@ -820,18 +826,23 @@ func (p *DataSource) convertToTableScan(prop *requiredProp) (task task, err erro
 	}.init(p.allocator, p.ctx)
 	ts.SetSchema(p.schema)
 	sc := p.ctx.GetSessionVars().StmtCtx
+	ts.Ranges = []types.IntColumnRange{{math.MinInt64, math.MaxInt64}}
+	pkColumn := expression.ColInfo2Col(ts.schema.Columns, ts.Table.GetPkColInfo())
 	if len(p.pushedDownConds) > 0 {
 		conds := make([]expression.Expression, 0, len(p.pushedDownConds))
 		for _, cond := range p.pushedDownConds {
 			conds = append(conds, cond.Clone())
 		}
-		ts.AccessCondition, ts.filterCondition = ranger.DetachColumnConditions(conds, p.tableInfo.GetPkName())
-		ts.Ranges, err = ranger.BuildTableRange(ts.AccessCondition, sc)
-		if err != nil {
-			return nil, errors.Trace(err)
+		if pkColumn != nil {
```

Comment:
the `GetPkColInfo` may return nil, it's not a good code style. Here we can check pk is handle and make sure pk column is not nil.